test/unittests/OpModel/TTNN/Lib/TestOpModelLib.cpp

```diff
@@ -0,0 +1,660 @@
+  std::tie(legal, l1Usage, errorMsg) = opConstraint;
+  EXPECT_TRUE(legal);
+  EXPECT_TRUE(l1Usage.has_value());
+  EXPECT_FALSE(errorMsg.has_value());
+  std::tie(cb_size, peak_size, output_size) = l1Usage.value();
+  EXPECT_EQ(cb_size, 786432);
+  EXPECT_EQ(output_size, 0);
+  EXPECT_EQ(peak_size, 0);
+
+  opConstraint = MatmulOpInterface::getOpConstraints(
+      tensorShape, inputLayout_dram, tensorShape, inputLayout_l1, tensorShape,
+      inputLayout_l1, false, false, workerGrid);
+  std::tie(legal, l1Usage, errorMsg) = opConstraint;
+  EXPECT_TRUE(legal);
+  EXPECT_TRUE(l1Usage.has_value());
+  EXPECT_FALSE(errorMsg.has_value());
+  std::tie(cb_size, peak_size, output_size) = l1Usage.value();
+  EXPECT_EQ(cb_size, 786432);
+  EXPECT_EQ(output_size, 151552);
+  EXPECT_EQ(peak_size, 151552);
+
+  opConstraint = MatmulOpInterface::getOpConstraints(
+      tensorShape, inputLayout_l1, tensorShape, inputLayout_dram, tensorShape,
+      inputLayout_dram, false, false, workerGrid);
+  std::tie(legal, l1Usage, errorMsg) = opConstraint;
+  EXPECT_TRUE(legal);
+  EXPECT_TRUE(l1Usage.has_value());
+  EXPECT_FALSE(errorMsg.has_value());
+  std::tie(cb_size, peak_size, output_size) = l1Usage.value();
+  EXPECT_EQ(cb_size, 786432);
+  EXPECT_EQ(output_size, 0);
+  EXPECT_EQ(peak_size, 0);
+
+  opConstraint = MatmulOpInterface::getOpConstraints(
+      tensorShape, inputLayout_l1, tensorShape, inputLayout_dram, tensorShape,
+      inputLayout_l1, false, false, workerGrid);
+  std::tie(legal, l1Usage, errorMsg) = opConstraint;
+  EXPECT_TRUE(legal);
+  EXPECT_TRUE(l1Usage.has_value());
+  EXPECT_FALSE(errorMsg.has_value());
+  std::tie(cb_size, peak_size, output_size) = l1Usage.value();
+  EXPECT_EQ(cb_size, 786432);
+  EXPECT_EQ(output_size, 151552);
+  EXPECT_EQ(peak_size, 151552);
+
+  opConstraint = MatmulOpInterface::getOpConstraints(
+      tensorShape, inputLayout_l1, tensorShape, inputLayout_l1, tensorShape,
+      inputLayout_dram, false, false, workerGrid);
+  std::tie(legal, l1Usage, errorMsg) = opConstraint;
+  EXPECT_TRUE(legal);
+  EXPECT_TRUE(l1Usage.has_value());
+  EXPECT_FALSE(errorMsg.has_value());
+  std::tie(cb_size, peak_size, output_size) = l1Usage.value();
+  EXPECT_EQ(cb_size, 786432);
+  EXPECT_EQ(output_size, 0);
+  EXPECT_EQ(peak_size, 0);
+
+  opConstraint = MatmulOpInterface::getOpConstraints(
+      tensorShape, inputLayout_l1, tensorShape, inputLayout_l1, tensorShape,
+      inputLayout_l1, false, false, workerGrid);
+  std::tie(legal, l1Usage, errorMsg) = opConstraint;
+  EXPECT_TRUE(legal);
+  EXPECT_TRUE(l1Usage.has_value());
+  EXPECT_FALSE(errorMsg.has_value());
+  std::tie(cb_size, peak_size, output_size) = l1Usage.value();
+  EXPECT_EQ(cb_size, 786432);
+  EXPECT_EQ(output_size, 151552);
+  EXPECT_EQ(peak_size, 151552);
+}
+
+TEST_F(OpModelBase, MatmulSharded) {
+  const llvm::SmallVector<int64_t> tensorShape = {1024, 1024};
+  const SmallVector<int64_t> gridShape = {4, 4};
```

Comment:
Is there a reason for not using 8x8 here?

Reply:
Matmul does check actual grid size and thus fails
 Number of rows in matmul compute grid exceeds maximum device compute grid size!


it works fine with 7x8